### ISSUE TITLE
[java] AbstractClassWithoutAbstractMethod false negative with inner abstract classes

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#2398]
**Affects PMD Version:** 6.22.0

**Rule:** [AbstractClassWithoutAbstractMethod](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#abstractclasswithoutabstractmethod)

**Description:**

When a abstract class contains a abstract inner class which has some abstract methods, then the outer abstract class is not detected, that it has no abstract methods.

**Code Sample demonstrating the issue:**

Test cases "Abstract class with inner classes false-negative 2" and "Abstract class with inner classes false-negative 3" are failing right now.

```xml
    <test-code>
        <description>Abstract class with inner classes false-negative 1</description>
        <expected-problems>1</expected-problems>
        <expected-linenumbers>1</expected-linenumbers>
        <code><![CDATA[
public abstract class Foo {
    public void concreteMethod() {}
    public abstract static class MyInner implements Other {
    }
}
        ]]></code>
    </test-code>

    <test-code>
        <description>Abstract class with inner classes false-negative 2</description>
        <expected-problems>1</expected-problems>
        <expected-linenumbers>1</expected-linenumbers>
        <code><![CDATA[
public abstract class Foo {
    public void concreteMethod() {}
    public abstract static class MyInner extends Other {
    }
}
        ]]></code>
    </test-code>

    <test-code>
        <description>Abstract class with inner classes false-negative 3</description>
        <expected-problems>1</expected-problems>
        <expected-linenumbers>1</expected-linenumbers>
        <code><![CDATA[
public abstract class Foo {
    public void concreteMethod() {}
    public abstract static class MyInner {
        public abstract void someMethod();
    }
}
        ]]></code>
    </test-code>

    <test-code>
        <description>Class with abstract inner class false-negative</description>
        <expected-problems>1</expected-problems>
        <expected-linenumbers>2</expected-linenumbers>
        <code><![CDATA[
public class Foo {
    public abstract static class MyInner {
        public void concreteMethod() {}
    }
}
        ]]></code>
    </test-code>
```

**Expected outcome:**

* This is a false negative

**Additional notes:**

* This is being fixed with the changes in #2377 .
